nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-services/src/main/java/org/apache/nifi/mongodb/MongoDBLookupService.java

```diff
@@ -52,68 +54,125 @@
     "The query is limited to the first result (findOne in the Mongo documentation). If no \"Lookup Value Field\" is specified " +
     "then the entire MongoDB result document minus the _id field will be returned as a record."
 )
-public class MongoDBLookupService extends MongoDBControllerService implements LookupService<Object> {
+public class MongoDBLookupService extends SchemaRegistryService implements LookupService<Object> {
+    public static final PropertyDescriptor CONTROLLER_SERVICE = new PropertyDescriptor.Builder()
+        .name("mongo-lookup-client-service")
+        .displayName("Client Service")
+        .description("A MongoDB controller service to use with this lookup service.")
+        .required(true)
+        .identifiesControllerService(MongoDBControllerService.class)
```

Comment:
I believe this is supposed to be an interface not the impl class (see my other comment below), so I think you want `MongoDBClientService` here.